htlcswitch/switch_test.go

```diff
@@ -1294,6 +1296,178 @@ type multiHopFwdTest struct {
 	expectedReply        lnwire.FailCode
 }
 
+// TestCircularForwards tests the allowing/disallowing of circular payments
+// through the same channel in the case where the switch is configured to allow
+// and disallow same channel circular forwards.
+func TestCircularForwards(t *testing.T) {
+	tests := []struct {
+		name                 string
+		allowCircularPayment bool
+		expectedFailure      lnwire.FailureMessage
+		expectedErrorString  string
+	}{
+		{
+			name:                 "circular payment allowed",
+			allowCircularPayment: true,
+			expectedFailure:      nil,
+		},
+		{
+			name:                 "circular payment disallowed",
+			allowCircularPayment: false,
+			expectedFailure:      lnwire.NewTemporaryChannelFailure(nil),
+			expectedErrorString:  fmt.Sprintf("%v", FailureDetailCircularRoute),
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			alicePeer, err := newMockServer(
+				t, "alice", testStartingHeight, nil, 6,
```

Comment:
I know this is a pattern in this file, but it really bothers me that 6 is not a constant :)

Reply:
nit and thy shall receive! (a separate commit declaring a constant for these tests 😄 )

Thanks for pointing it out, it's always nice to do a little clean up while you're around. I think it would be cool to switch over to functional options for some or these test structs with a sensible default so that exactly what you're spinning up is more clear/controlled. But that's a separate discussion/PR.